### PR TITLE
Add DataFrame.fromArrow

### PR DESCRIPTION
**Pull request by bryevdv**
Add a static `DataFrame.fromArrow` method that can adapt Arrow tables stored in-memory in Arrow IPC format.
 
This is still WIP but early comments welcome. Some notes:

* As implemented, this requires a copy. Plan to to push improvements upstream to `cudf::from_arrow` to allow it to perform zero-copy when feasible (e.g. ipc buffers)
* Tests are a bit thin. Open to suggestions to improve coverage parsimoniously 
* ~~Still need to add an input type to `fromArrow` (similar to `readCSV`) so that full types can be deduced~~

### CODE REVIEW

File: modules/cudf/src/data_frame.ts [hunk 1]
```diff
@@ -103,6 +104,18 @@ export class DataFrame<T extends TypeMap = any> {
                    {} as ColumnsMap<{[P in keyof T]: CSVToCUDFType<T[P]>}>)));
   }
 
+  /**
+   * Reads a an arrow IPC table into a DataFrame.
```
Comment (mzegar) [inline modules/cudf/src/data_frame.ts:108]: Grammar here `Reads a an arrow` -> `Reads an arrow`

File: modules/cudf/src/table.ts [hunk 2]
```diff
@@ -45,6 +47,14 @@ interface TableConstructor {
   readCSV<T extends CSVTypeMap = any>(options: ReadCSVOptions<T>):
     {names: (keyof T)[], table: Table};
 
+  /**
+   * Reads a an arrow IPC table into a set of columns.
```
Comment (mzegar) [inline modules/cudf/src/table.ts:51]: Same as comment above.

File: modules/cudf/src/table/arrow.cpp [hunk 3]
```diff
@@ -57,4 +58,41 @@ Napi::Value Table::to_arrow(Napi::CallbackInfo const& info) {
...
+  std::shared_ptr<arrow::Table> arrow_table{};
+  auto status = stream_reader->ReadAll(&arrow_table);
+
+  if (!status.ok()) { NAPI_THROW(Napi::Error::New(env, status.message())); }
+
+  auto output = Napi::Object::New(env);
+
+  auto fields = stream_reader->schema()->fields();
+  auto names  = Napi::Array::New(env, fields.size());
+  for (std::size_t i = 0; i < fields.size(); ++i) { names.Set(i, fields[i]->name()); }
```
Comment (mzegar) [inline modules/cudf/src/table/arrow.cpp:87]: Is there a `std` operations for this? `transform`? Regardless, this still looks clean.

**Review by mzegar — APPROVED** [review]
Initial comments

File: modules/cudf/src/table/arrow.cpp (hunk 3 above)
Comment (bryevdv) [reply modules/cudf/src/table/arrow.cpp:87]: We would need an NAPI approach that didn't require the iteration index, or `std` operation that provides one (i.e. like Python's `enumerate`) but AFAIK I don't think either exists.

**Comment by mzegar** [comment]
Reviewed in the morning and didn't realize I accidently approved. Will give it another look over.

File: modules/cudf/src/table/arrow.cpp (hunk 3 above)
Comment (trxcllnt) [reply modules/cudf/src/table/arrow.cpp:87]: It'd be nice if `Napi::Array` had an iterator similar to https://github.com/nodejs/node-addon-api/pull/930

**Comment by mzegar** [comment]
`BINDINGS.md` needs to be updated, otherwise, got nothing else to comment Might be good to get another review :+1:

**Comment by bryevdv** [comment]
Good catch, updated!

**Pull request merged**